src/print.jl

```diff
@@ -324,6 +326,175 @@ function objective_function_string(print_mode, model::Model)
    end
 end
 
+struct _SolutionSummary
+    verbose::Bool
+    solver::String
+    # Status
+    termination_status::MOI.TerminationStatusCode
+    primal_status::MOI.ResultStatusCode
+    dual_status::MOI.ResultStatusCode
+    raw_status::String
+    result_count::Int
+    has_values::Bool
+    has_duals::Bool
+    # Candidate solution
+    objective_value::Float64
+    objective_bound::Union{Missing,Float64}
+    dual_objective_value::Union{Missing,Float64}
+    primal_solution::Dict{String,Float64}
+    dual_solution::Dict{String,Float64}
+    # Work counters
+    solve_time::Float64
+    barrier_iterations::Union{Missing,Int}
+    simplex_iterations::Union{Missing,Int}
+    node_count::Union{Missing,Int}
+end
+
+"""
+    solution_summary(model::Model; verbose::Bool = false)
+
+Return a struct that can be used print a summary of the solution.
+
+If `verbose=true`, write out the primal solution for every variable and the
+dual solution for every constraint, excluding those with empty names.
+
+## Examples
+
+When called at the REPL, the summary is automatically printed:
+```julia
+julia> solution_summary(model)
+[...]
+```
+
+Use `print` to force the printing of the summary from inside a function:
+```julia
+function foo(model)
+    print(solution_summary(model))
+    return
+end
+```
+"""
+solution_summary(model::Model; verbose::Bool = false) = _SolutionSummary(
+    verbose,
+    solver_name(model),
+    termination_status(model),
+    primal_status(model),
+    dual_status(model),
+    raw_status(model),
+    result_count(model),
+    has_values(model),
+    has_duals(model),
+    objective_value(model),
+    _try_get(objective_bound, model),
+    _try_get(dual_objective_value, model),
```

Comment:
We should have an array here with one entry for `i in 1:result_count` that contains a struct containing the objective value, solution and dual solution, dual objective value, ...

Reply:
Do you have an example code which has result_count is 2 or more? I would like to add a test using it.

Reply:
I found this codes. Maybe I can use it. https://github.com/jump-dev/JuMP.jl/blob/90e63c47372e02b7093a9bf2fb4153dfa548c0a6/test/generate_and_solve.jl#L601-L638

Reply:
Let's just leave this as-is for now. It covers 99.9% of peoples use-cases, and I'm mindful that @AtsushiSakai has put a lot of effort in, and we keep asking for more and more changes.

In future, we could add a `alternative_solutions::Vector` that is backward compatible. The first result is special as it is the only solution guaranteed to be optimal if `TerminationStatus` is optimal.